bgpd/bgp_vty.c

```diff
@@ -6112,6 +6112,525 @@ ALIAS_HIDDEN(no_neighbor_addpath_tx_bestpath_per_as,
+				   bgp_get_default(), bgp);
+
+		bgp->vpn_policy[afi].tovpn_label = MPLS_LABEL_NONE;
+
+		/* post-change: re-export vpn routes */
+		vpn_leak_postchange(BGP_VPN_POLICY_DIR_TOVPN, afi,
+				    bgp_get_default(), bgp);
+	}
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (vpn_policy_nexthop,
+       vpn_policy_nexthop_cmd,
+       "nexthop <A.B.C.D|X:X::X:X>$nexthop",
+       "Specify next hop to use for VRF advertised prefixes\n"
+       "IPv4 prefix\n"
+       "IPv6 prefix\n")
+{
+	VTY_DECLVAR_CONTEXT(bgp, bgp);
+	int doafi[AFI_MAX] = {0};
+	afi_t afi;
+	int ret;
+	struct prefix p;
+
+	if (!sockunion2hostprefix(nexthop, &p)) 
+		return CMD_WARNING_CONFIG_FAILED;
+
+	ret = vpn_policy_afis(vty, doafi);
+	if (ret != CMD_SUCCESS)
+		return ret;
+
+	for (afi = 0; afi < AFI_MAX; ++afi) {
+		if (!doafi[afi])
+			continue;
+
+		/*
+		 * pre-change: un-export vpn routes (vpn->vrf routes unaffected)
+		 */
+		vpn_leak_prechange(BGP_VPN_POLICY_DIR_TOVPN, afi,
+				   bgp_get_default(), bgp);
+
+		bgp->vpn_policy[afi].tovpn_nexthop = p;
+		SET_FLAG(bgp->vpn_policy[afi].flags,
+			 BGP_VPN_POLICY_TOVPN_NEXTHOP_SET);
+
+		/* post-change: re-export vpn routes */
+		vpn_leak_postchange(BGP_VPN_POLICY_DIR_TOVPN, afi,
+				    bgp_get_default(), bgp);
+	}
+
+	return CMD_SUCCESS;
+}
+
+DEFUN (vpn_policy_no_nexthop,
+       vpn_policy_no_nexthop_cmd,
+       "no nexthop",
+       NO_STR
+       "Specify next hop to use for VRF advertised prefixes\n")
+{
+	VTY_DECLVAR_CONTEXT(bgp, bgp);
+	int doafi[AFI_MAX] = {0};
+	afi_t afi;
+	int ret;
+
+	ret = vpn_policy_afis(vty, doafi);
+	if (ret != CMD_SUCCESS)
+		return ret;
+
+	for (afi = 0; afi < AFI_MAX; ++afi) {
+		if (!doafi[afi])
+			continue;
+
+		/* pre-change: un-export vpn routes (vpn->vrf routes unaffected)
+		 */
+		vpn_leak_prechange(BGP_VPN_POLICY_DIR_TOVPN, afi,
+				   bgp_get_default(), bgp);
+
+		UNSET_FLAG(bgp->vpn_policy[afi].flags,
+			   BGP_VPN_POLICY_TOVPN_NEXTHOP_SET);
+
+		/* post-change: re-export vpn routes */
+		vpn_leak_postchange(BGP_VPN_POLICY_DIR_TOVPN, afi,
+				    bgp_get_default(), bgp);
+	}
+
+	return CMD_SUCCESS;
+}
+
+static int set_ecom_list(struct vty *vty, int argc, struct cmd_token **argv,
+			 struct ecommunity **list)
+{
+	struct ecommunity *ecom = NULL;
+	struct ecommunity *ecomadd;
+
+	for (; argc; --argc, ++argv) {
+
+		ecomadd = ecommunity_str2com(argv[0]->arg,
+					     ECOMMUNITY_ROUTE_TARGET, 0);
+		if (!ecomadd) {
+			vty_out(vty, "Malformed community-list value\n");
+			if (ecom)
+				ecommunity_free(&ecom);
+			return CMD_WARNING_CONFIG_FAILED;
+		}
+
+		if (ecom) {
+			ecommunity_merge(ecom, ecomadd);
+			ecommunity_free(&ecomadd);
+		} else {
+			ecom = ecomadd;
+		}
+	}
+
+	if (*list) {
+		ecommunity_free(&*list);
+	}
+	*list = ecom;
+
+	return CMD_SUCCESS;
+}
+
+DEFUN (vpn_policy_rt,
+       vpn_policy_rt_cmd,
+       "rt <fromvpn|tovpn|both> RTLIST...",
+       "Specify route target list\n"
+       "fromvpn: match any\n"
```

Comment:
This doesn't make sense to me. Is it intended syntax or is "match any" describing the action of the keyword? It seems to be syntax as referenced on line 6445 but I am not sure.